app/controllers/songs_controller.rb

```diff
@@ -1,5 +1,34 @@
 class SongsController < ApplicationController
   def index
+    respond_to do |format|
+      format.json do
+        songs = Song.all
+        total_songs = songs.count
+
+        if params[:sSearch].present?
+          search = {
+            query: "%#{params[:sSearch]}%",
+            rQuery: Regexp.escape(params[:sSearch])
+          }
+
+          songs = songs.where('name LIKE :query OR artist LIKE :query OR
```

Comment:
Having `songs = Song.all` and then `songs.where()` here seems redundant. You can probably combine it into `songs = Song.where()` here can't you I think. Also, it seems this query logic should be abstracted away for isolation and DRYness. I'll work on that.